src/devices/tuya.ts

```diff
@@ -7670,34 +7930,49 @@ const definitions: Definition[] = [
         model: 'PJ-1203A',
         vendor: 'TuYa',
         description: 'Bidirectional energy meter with 80A current clamp',
-        fromZigbee: [tuya.fz.datapoints],
+        fromZigbee: [PJ1203A.strict_fz_datapoints, PJ1203A.sync_time_increase_seq],
         toZigbee: [tuya.tz.datapoints],
         onEvent: tuya.onEventSetTime,
         configure: tuya.configureMagicPacket,
+        options: [
+            PJ1203A.options.late_energy_flow('a'),
+            PJ1203A.options.late_energy_flow('b'),
+            PJ1203A.options.signed_power('a'),
+            PJ1203A.options.signed_power('b'),
+        ],
         exposes: [
             e.ac_frequency(), e.voltage(),
-            tuya.exposes.powerWithPhase('a'), tuya.exposes.powerWithPhase('b'), tuya.exposes.powerWithPhase('ab'),
-            tuya.exposes.currentWithPhase('a'), tuya.exposes.currentWithPhase('b'),
-            tuya.exposes.powerFactorWithPhase('a'), tuya.exposes.powerFactorWithPhase('b'),
-            tuya.exposes.energyFlowWithPhase('a'), tuya.exposes.energyFlowWithPhase('b'),
-            tuya.exposes.energyWithPhase('a'), tuya.exposes.energyWithPhase('b'),
-            tuya.exposes.energyProducedWithPhase('a'), tuya.exposes.energyProducedWithPhase('b'),
-            e.numeric('update_frequency', ea.STATE).withUnit('s').withDescription('Update frequency'),
+            tuya.exposes.powerWithChannel('a'), tuya.exposes.powerWithChannel('b'), tuya.exposes.powerWithChannel('ab'),
```

Comment:
Why not use `powerWithPhase` here?

Reply:
This is because a Phase and a Channel are two different things from an electrical point of view. The  PJ-1203A is a dual channel (or 2 gangs) device vs for instance the `TS0601_3_phase_clamp_meter` that is a 3-phase meter. 
That does not change anything from the perspective of Z2M. I was just a bit annoyed to see both `phase` and `device` in the documentation.

Another way could be to keep Phase in the function name but with an optional argument to customize the documentation. Something like that should work I think:
```javascript
    powerWithPhase: (phase: string, kind: string = 'phase') => e.numeric(`power_${phase}`, ea.STATE).withUnit('W')
        .withDescription(`Instantaneous measured power (${kind} ${phase.toUpperCase()})`),
```
and then
```
 tuya.exposes.powerWithPhase('a', kind='channel')
```

Reply:
Looks good! Maybe call it `powerWithPostifx` then?

Reply:
But then I would have to do the renaming it in all devices. Also that would also break a lot of external converters.

More generally, if we go that way the converter could simply be `power` with an optional argument `suffix`. 
And some helper functions could be added to automate the process. Something like that 

```javascript
function extend(name,suffix) { 
   return suffix ? name+"_"+suffix : name ;
}

function group(kind: string, name: string) {
  return name ? ` (${kind} ${name})` :  "";
} 

power: (suffix: string = null, kind: string = 'phase') => e.numeric(extend("power",suffix), ea.STATE).withUnit('W')
        .withDescription("Instantaneous measured power"+group(kind,suffix.toUpperCase()) ,

```

Reply:
Even better!

Reply:
The proposed changes would impact several converters so, for now, I am switching back to the original `WithPhase` syntax.  Maybe for another pull request  ...